nimble/host/src/ble_l2cap_coc.c

```diff
@@ -232,21 +311,28 @@ ble_l2cap_coc_rx_fn(struct ble_l2cap_chan *chan)
 }
 
 struct ble_l2cap_chan *
-ble_l2cap_coc_chan_alloc(uint16_t conn_handle, uint16_t psm, uint16_t mtu,
+ble_l2cap_coc_chan_alloc(struct ble_hs_conn *conn, uint16_t psm, uint16_t mtu,
                          struct os_mbuf *sdu_rx, ble_l2cap_event_fn *cb,
                          void *cb_arg)
 {
     struct ble_l2cap_chan *chan;
+    int scid;
 
-    chan = ble_l2cap_chan_alloc(conn_handle);
+    if((scid = ble_l2cap_coc_get_cid(conn)) == -1) {
+        /* No Free channel, can't allocate */
+        return NULL;
+    }
+
+    chan = ble_l2cap_chan_alloc(conn->bhc_handle);
     if (!chan) {
```

Comment:
since number of CIDs is related to the channels mempool, if CID can be allocated then channel shall be. Therefore, `assert` instead of `if (!chan)` should be sufficient.

Reply:
fixed